openlibrary/templates/book_providers/direct_read_button.html

```diff
@@ -0,0 +1,30 @@
+$def with(provider)
+
+$if provider.access == 'open-access':
+  <div class="cta-button-group">
+    <a
+        href="$(provider.url)"
+        title="$_('Read free online')"
+        class="cta-btn cta-btn--available cta-btn--read cta-btn--external cta-btn--direct"
+        target="_blank"
+        aria-haspopup="true"
+        aria-controls="direct-provider-toast"
+    >$_('Read')</a>
+  </div>
+
+$elif provider.access == 'sample':
+  <div class="cta-button-group">
+    <a class="cta-btn cta-btn--shell cta-btn--external"
+      data-ol-link-track="CTAClick|Preview"
+      target="_blank"
+      href="$(provider.url)"
+    >$_('Preview')</a>
+  </div>
+
+$if render_once('direct-provider-toast'):
+  <div class="toast toast--book-provider" data-toast-trigger=".cta-btn--direct" id="direct-provider-toast" style="display:none">
+    <div class="toast__body">
+      $:_('This book is available from <a href="%s">%s</a>, a site external to Open Library. By visiting this external site, you will be subject to that site\'s privacy policy and terms of use. Open Library cannot make any guarantees about this external site\'s content.', provider.url, provider.url)
```

Comment:
Inside the `<a>` can we parse the URL and display just the hostname? e.g. `eloquentjavascript.net` ?

```suggestion
      $:_('This book is freely available from <a href="%s">%s</a>, an external third-party book provider.', provider.url, domain)
      <a href="/trusted-book-providers#web-books">$_('Learn more')<a>
```